### DISCUSSION TITLE
I am getting error while generating token and using the same in subsquent API requests

### DISCUSSION THREAD

**Discussion by shiv-vsics** [artilleryio/artillery#2185] [Artillery]
Hi,
Could you please help me in this may be?
I have having one query related to how to generate token and then use the same token in subsequent request.

Below is given sample file code , i want to generate token from one token endpoint and then want to use the same in API endpoint. In the script which i written is givnig me error like:

return done(new Error('Failed capture or match'), context);
Error: Failed capture or match
[test1.txt](https://github.com/artilleryio/artillery/files/12805249/test1.txt)

[test1.txt](https://github.com/artilleryio/artillery/files/12802432/test1.txt)

Could you please help me it ? like why i am getting this error ?

**Comment by InesNi** [comment]
Hi @shiv-vsics

From the test you shared it seems the following part is where the issue happens:
```yaml
  capture:
    - json: $.access_token
      as: token
```
Can you check if the response from your API returns the token you are trying to capture in the exact format you are trying to capture it in? So it should be returning a JSON object with `access_token` key.

**Reply by shiv-vsics** [reply]
@InesNi Apologies for late response. This is correct only in my token generation API , response is coming in JSON format only like below: Any suggestion for the same?

**API Response is something like below:**
**{
    "token_type": "Bearer",
    "expires_in": 3600,
    "access_token": "iLCJQTEFQSSIsIk1ERCIsIlNQR0xPQkFMIiwiRVRBIiwiU0MiLCJDUyIsIlBMTURWIiwiUExUU1VCUyIsIlBMQ09NIiwiTUtUUEwiLCJTUEdQTEFUVFMiLCJNSSIsIkRJTUVOU0lPTlMiLCJCRU5QT1JUIiwiUE1DIl0sImNsaWVudCI6IlBMX0FQSV9QTEFURk9STSIsImNvbXBhbnkiOlsiQmhhdGlhLCBSYW1hbmRlZXAiXX0iuoweiruwoeiruwoeiruwreowierowueirw",
    "scope": "openid profiles test api",
    "refresh_token": "O_iuweriuwbaksjhasdHKJJuebdkjUHjHJhKJkJJHiduheknnekjEBdn",
    "id_token": "LCJNSSIsIkRJTUVOU0lPTlMiLCJCRU5QT1JUIiwiUE1DIl0sImNsaWVudCI6IlBMX0FQSV9QTEFURk9STSIsImNvbXBhbnkiOlsiQmhhdGlhLCBSYW1hbmRlZXAiXX0iuoweiruwoeiruwoeiruwreowierowueirw"
}**

**Comment by KKardy** [comment]
I get the same error. Please answer.
